pkg/kubelet/network/hostport/hostport_manager.go

```diff
@@ -152,6 +156,12 @@ func (hm *hostportManager) Add(id string, podPortMapping *PodPortMapping, natInt
 		// clean up opened host port if encounter any error
 		return utilerrors.NewAggregate([]error{err, hm.closeHostports(hostportMappings)})
 	}
+
+	// Ensure the chain is linked from the root.
+	if err = ensureKubeHostportChainLinked(hm.iptables); err != nil {
```

Comment:
yeah I think you're right.  iptables-restore will reject attempts to -j to a chian that's not in the restore output and/or not yet created, so the syncIPTables() call would fail if ensureKubeHostportChain() failed at creation time.

However, you could add another EnsureChain(kubeHostportsChain) near the top of Add() and get around that.

But I'm also curious why this was moved down below...